azure-kusto-data/azure/kusto/data/_telemetry.py

```diff
@@ -97,58 +97,29 @@
         return cluster_attributes
 
 
-class KustoTracing:
-    @staticmethod
-    def call_func_tracing(func: Callable, *args, **kwargs):
-        """
-        Prepares function for tracing and calls it
-        :param func: function to trace
-        :type func: Callable
-        :key str name_of_span: name of the trace span
-        :key dict tracing_attributes: key/value dictionary of attributes to include in span of trace
-        :key str kind: the type of span
-        :param kwargs: function arguments
-        """
-        name_of_span: str = kwargs.pop("name_of_span", None)
-        tracing_attributes: dict = kwargs.pop("tracing_attributes", {})
-        kind: str = kwargs.pop("kind", SpanKind.CLIENT)
-
-        kusto_trace: Callable = distributed_trace(name_of_span=name_of_span, tracing_attributes=tracing_attributes, kind=kind)
-        kusto_func: Callable = kusto_trace(func)
-        return kusto_func(*args, **kwargs)
+class Span:
+    """
+    Span class for telemetry
+    """
 
     @staticmethod
-    async def call_func_tracing_async(func: Callable, *args, **kwargs):
+    def run(invoker: Callable, name_of_span: str = None, tracing_attributes=None, kind: str = SpanKind.INTERNAL):
```

Comment:
Use generics to signify that the Method returns what the callable does, like in the example I showed you in teams.

Reply:
addressed